security/pkg/pki/ca/ca.go

```diff
@@ -63,6 +63,27 @@ var pkiCaLog = log.RegisterScope("pkica", "Citadel CA log", 0)
 // caTypes is the enum for the CA type.
 type caTypes int
 
+type CertOpts struct {
+	// SubjectIDs are used for building the SAN extension for the certificate.
+	SubjectIDs []string
+
+	// TTL is the requested lifetime (Time to live) to be applied in the certificate.
+	TTL time.Duration
+
+	// ForCA indicates whether the signed certificate if for CA.
+	// If true, the signed certificate is a CA certificate, otherwise, it is a workload certificate.
+	ForCA bool
+}
+
+// NewCertOpts creates new cert options based on the given inputs.
+func NewCertOpts(subjectIDs []string, ttl time.Duration, forCA bool) *CertOpts {
```

Comment:
I am not sure this is a useful function, why not just make the `CertOpts` struct directly? Then you can use named fields and not include all of them?

Reply:
Make sense to me. Updated, thanks!